src/Files.App/Helpers/NavigationHelpers.cs

```diff
@@ -261,9 +262,16 @@ private static async Task<FilesystemResult> OpenDirectory(string path, IShellPag
 
 			var opened = (FilesystemResult)false;
 			bool isHiddenItem = NativeFileOperationsHelper.HasFileAttribute(path, System.IO.FileAttributes.Hidden);
-			bool IsShortcut = path.EndsWith(".lnk", StringComparison.Ordinal) || path.EndsWith(".url", StringComparison.Ordinal); // Determine
+			bool isShortcut = path.EndsWith(".lnk", StringComparison.Ordinal) || path.EndsWith(".url", StringComparison.Ordinal); // Determine
 
-			if (IsShortcut)
+			// Check for folder permission
+			if (!FolderHelpers.CheckFolderAccessWithWin32(path))
+			{
+				await DialogDisplayHelper.ShowDialogAsync(DynamicDialogFactory.GetFor_FolderNoPermission());
+				return (FilesystemResult)false;
+			}
```

Comment:
Do we have to explicitly check for access? I'm pretty sure the navigation apis will throw for access denied.

Reply:
Good point. I’ll check this.

Reply:
I can't get my head around it. If there is an exception, is must be handled somewhere already. I guess we don't want to be checking each folder before we open them if the user has permission or not though.

Reply:
cc @gave92

Reply:
Sry missed this comment. I've just checked with `C:\Users\Administrator` and the is actually no exception being thrown (ItemViewModel/EnumerateItemsFromStandardFolderAsync() would be the place). The Storage API does not return any error, it just makes the folder appear empty. I'm positive this was not the case with UWP, we even have code for showing a dialog:
```cs
else if (res == FileSystemStatusCode.Unauthorized)
{
  //TODO: proper dialog
  await DialogDisplayHelper.ShowDialogAsync(
    "AccessDenied".GetLocalizedResource(),
    "SubDirectoryAccessDenied".GetLocalizedResource());
  return -1;
}
```

Reply:
Using `CheckFolderAccessWithWin32` is probably not ok because I imagine it would show access denied when browsing e.g phone storage.